components/Common/Search/States/WithSearchBox.tsx

```diff
@@ -135,14 +126,17 @@ export const WithSearchBox: FC<SearchBoxProps> = ({ onClose }) => {
     router.push(`/search?q=${searchTerm}&section=${selectedFacetName}`);
   };
 
-  const changeFacet = (idx: string) => setSelectedFacet(Number(idx));
+  const changeFacet = (idx: string) => {
```

Comment:
Instead of casting the "idx" twice as a Number, why not cast it within the calls of changeFacet?

Reply:
would not make much of a difference tho

Reply:
It would, as it would explicitly tell that a facet must be a number. Otherwise I can pass "blah blah" as a param a a number conversion would be NaN; It enforces that whoever is calling the method is responsible for guaranteeing a number.

Reply:
`<Tabs
              activationMode="manual"
              defaultValue="0"
              autoFocus={true}
              tabs={Object.keys(facets).map((facetName, idx) => ({
                key: facetName,
                label: facetName,
                secondaryLabel: `(${facets[facetName].toLocaleString('en')})`,
                value: idx.toString(),
              }))}
              onValueChange={changeFacet}
            />`
            
            changeFacet is a function that is being passed to the Tabs component.Its type must be `(value: string) => void` .

Reply:
`onValueChange={changeFacet}` changeFacet is being passed in the Tabs component.Its type must be `(value: string) => void`.